modules/registry/registry.go

```diff
@@ -203,3 +253,67 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 
 	return c, nil
 }
+
+// SetDockerAuthConfig sets the DOCKER_AUTH_CONFIG environment variable with
+// authentication for with the given host, username and password.
+// It returns the base64 encoded credentials and a function to reset back
+// to the previous state.
+func SetDockerAuthConfig(host, username, password string, additional ...string) (creds string, cleanup func(), err error) { //nolint:nonamedreturns // Adds context to the return values.
+	creds, auth, err := DockerAuthConfig(host, username, password, additional...)
+	if err != nil {
+		return "", nil, fmt.Errorf("docker auth config: %w", err)
+	}
+
+	previousAuthConfig := os.Getenv("DOCKER_AUTH_CONFIG")
+	os.Setenv("DOCKER_AUTH_CONFIG", auth)
+
+	cleanup = func() {
+		if previousAuthConfig == "" {
+			os.Unsetenv("DOCKER_AUTH_CONFIG")
+			return
+		}
+		os.Setenv("DOCKER_AUTH_CONFIG", previousAuthConfig)
+	}
+
+	return creds, cleanup, nil
+}
+
+// DockerAuthConfig returns the base64 encoded credentials and the auth config
+// for the provided details. It also accepts additional host, username and password
+// triples to add more auth configurations.
+func DockerAuthConfig(host, username, password string, additional ...string) (creds, authConfig string, err error) { //nolint:nonamedreturns // Adds context to the return values.
+	if username != "" || password != "" {
+		creds = base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+	}
+
+	cfg := dockercfg.Config{
+		AuthConfigs: map[string]dockercfg.AuthConfig{
+			host: {
+				Username: username,
+				Password: password,
+				Auth:     creds,
+			},
+		},
+		CredentialsStore: "desktop",
```

Comment:
I have tested without and still seems to work, but I'd be interested if anyone knows of any hidden issues that might create?

Reply:
I am fairly convinced, the `CredentialsStore` key here would not have any effect (unless providing meta information to Docker downstream?), since the actual credentials are already provided (right?). This should normally indicate to Docker, after reading the `config.json`, that it needs to load credentials using a desktop credential store helper. But this situation does not apply to this part of the code, right?

So removing it sounds right.

Reply:
Agreed, have removed and refactored the new methods a little bit to simplify given the original returns aren't needed.